src/lib/Key.js

```diff
@@ -4,10 +4,13 @@ class Key {
     /**
      * @param {Uint8Array} secret
      * @param {Key.Type} [type]
+     * @param {boolean} [hasPin]
      */
-    constructor(secret, type = Key.Type.BIP39) {
+    constructor(secret, type = Key.Type.BIP39, hasPin = false) {
         this._secret = secret;
         this._type = type;
+        /** @type {boolean} */
+        this._hasPin = !!hasPin;
```

Comment:
are the double exclamation marks required? Looks to me it is typed correctly so no need to force the boolean?

It does force the boolean as some sort of input validation which generally is never a bad thing, however I think there is no need for it here (also the other inputs aren't validated as well).

Curious what the others think.

Reply:
ha, same thought

Reply:
ha, same thought